### PR TITLE
Add environment variables to change max file sizes

### PR DESCRIPTION
**Pull request by PostPollux**
This pull request adds some environment variables to easily configure the maximum file size allowed in directus without the need to create a custom docker image just to change that.

Therefore a new entrypoint.sh script (api container) changes some values of php.ini and default.conf on startup depending on the environment variables.

### CODE REVIEW

**Comment by PostPollux** [comment]
Sorry, I don't know what s6 is. Any hint or helpful link, so i can research it?

**Comment by WoLfulus** [comment]
Since we have both nginx and fpm processes running, s6 manages them to bring down the container if something gets messed up.
ref: https://github.com/just-containers/s6-overlay

**Comment by PostPollux** [comment]
Thanks for the link, I will look into it when I find some time.
Basically, do you think it's a good idea to add those variables? Because if not, I'm not sure if it's worth the time to dig deeper. :)

**Comment by WoLfulus** [comment]
I think it is since we have a feature directly related to that. But this will probably bring problems if someone just wants to map the whole file (will overwrite the values from the file) - checking for env variable existance before running sed instead of defaulting the value would solve this though)

**Comment by PostPollux** [comment]
Ok, I looked a bit into the very basics of s6.
So this would mean I put this sed stuff into those two run scripts?
- docker/projects/api/rootfs/etc/services.d/php_fpm/run
- docker/projects/api/rootfs/etc/services.d/nginx/run

**Comment by PostPollux** [comment]
Any Feedback? Actually I did not test my last force-push as I don't know how to build the image. Building a Docker image is normally not that complicated, but in case of the directus docker repository with s6, separation with a builder image and automatic webhook scripts I'm a bit overwhelmed. How do I just build the api image locally with 2.0.18 tag for testing without uploading something to dockerhub?

**Comment by fl034** [comment]
Keep up with your good work in making often needed settings like this more accessible.

I ended up mounting a `custom.ini` file in my `docker-compose.yaml` like this:

```
  directus-api:
    image: directus/api:latest
    volumes:
      - ./config/directus-api/php/custom.ini:/etc/php7/conf.d/custom.ini
```

with the following content:
```
memory_limit = 64M
upload_max_filesize = 64M
post_max_size = 64M
max_execution_time = 600
```

So I have 64MB upload limit without the need of creating a custom docker image.

**Comment by WoLfulus** [comment]
I'm working on rewriting the whole repository because of what you said @PostPollux 

We're still trying to figure out a way to make it easy to change settings @fl034 , any suggestions are appreciated.

File: projects/api/rootfs/etc/services.d/php_fpm/run [hunk 1]
```diff
@@ -1,5 +1,35 @@
 #!/usr/bin/with-contenv sh
 set -e;
 
+
+# Apply some environment variables if they are set
+
+# set maximum upload file size (default is 2M)
+if ! [[ -z "${UPLOAD_MAX_FILESIZE}" ]]; then
+  sed -i "s/upload_max_filesize =.*/upload_max_filesize = ${UPLOAD_MAX_FILESIZE}/g" /etc/php7/php.ini
```
Comment (mvhirsch) [inline projects/api/rootfs/etc/services.d/php_fpm/run]: Why not just `echo` them to `/etc/php7/conf.d/20_directus.ini`?

**Comment by mvhirsch** [comment]
This looks very promising and indeed would fix a common needed problem, too.
My two cents on this would be prefixing the `PHP_` envs, so it's clear what they do.

**Comment by WoLfulus** [comment]
We've merged the new repository structure. This doesn't apply anymore.

**Pull request closed without merging**